src/include/OpenImageIO/missing_math.h

```diff
@@ -29,132 +29,12 @@
 */
 
 
-/// \file
-///
-/// This adds definitions we think *should* be in math.h, but which on
-/// on some platforms appears to be missing in action.
-///
+// DEPRECATED(2.0)
+// Once we were fully C++11 and moved MSVS minimum to >= 2015, the contents
+// of this file were no longer needed. What few things were here that
+// anybody might still rely on, are in fmath.h.
 
 
 #pragma once
 
-#include <cmath>
-
-#include <OpenImageIO/oiioversion.h>
-
-
-#ifndef M_PI
-/// PI
-#    define M_PI 3.14159265358979323846264338327950288
-#endif
-
-#ifndef M_PI_2
-/// PI / 2
-#    define M_PI_2 1.57079632679489661923132169163975144
-#endif
-
-#ifndef M_PI_4
-/// PI / 4
-#    define M_PI_4 0.785398163397448309615660845819875721
-#endif
-
-#ifndef M_TWO_PI
-/// PI * 2
-#    define M_TWO_PI (M_PI * 2.0)
-#endif
-
-#ifndef M_1_PI
-/// 1/PI
-#    define M_1_PI 0.318309886183790671537767526745028724
-#endif
-
-#ifndef M_2_PI
-/// 2/PI
-#    define M_2_PI 0.636619772367581343075535053490057448
-#endif
-
-#ifndef M_SQRT2
-/// sqrt(2)
-#    define M_SQRT2 1.41421356237309504880168872420969808
-#endif
-
-#ifndef M_SQRT1_2
-/// 1/sqrt(2)
-#    define M_SQRT1_2 0.707106781186547524400844362104849039
-#endif
-
-#ifndef M_LN2
-/// ln(2)
-#    define M_LN2 0.69314718055994530941723212145817656
-#endif
-
-#ifndef M_LN10
-/// ln(10)
-#    define M_LN10 2.30258509299404568401799145468436421
-#endif
-
-#ifndef M_E
-/// e, Euler's number
-#    define M_E 2.71828182845904523536028747135266250
-#endif
-
-#ifndef M_LOG2E
-/// log2(e)
-#    define M_LOG2E 1.44269504088896340735992468100189214
-#endif
-
-
-OIIO_NAMESPACE_BEGIN
-
-#ifdef _WIN32
-// Windows doesn't define these functions from math.h
-#    define hypotf _hypotf
-#    define copysign(x, y) _copysign(x, y)
-#    define copysignf(x, y) float(copysign(x, y))
-
-
-inline float
-truncf(float val)
-{
-    return (float)(int)val;
-}
-
-
-inline float
-exp2f(float val)
-{
-    // 2^val = e^(val*ln(2))
-    return (float)exp(val * M_LN2);
-}
-
-
-#endif /* _WIN32 */
-
-
-#if OIIO_MSVS_AT_LEAST_2013 && __cplusplus <= 201103L
-// Prior to c++11, these were implementation defined, and on msvc, were
-// not in the std namespace.
-using ::isfinite;
-using ::isinf;
-using ::isnan;
-#else
-using std::isfinite;
-using std::isinf;
-using std::isnan;
-#endif
-
-
-
-// Functions missing from FreeBSD
-#if (defined(__FreeBSD__) && (__FreeBSD_version < 803000))
-
-inline float
-log2f(float val)
-{
-    return logf(val) / static_cast<float>(M_LN2);
-}
-
-#endif
-
-
-OIIO_NAMESPACE_END
+#include <OpenImageIO/fmath.h>
```

Comment:
Adding this line broke then OSL build because `simd.h` needs `missing_math.h` which needs `fmath.h` which needs `simd.h` (which hasn't yet been included).

Is it really necessary to bring in `fmath.h` here? or should it be `cmath` instead?

Reply:
I think it helps, because some of those #defines moved to fmath.h. 

The culprit here was that I hadn't removed stray include of missing_math.h from simd.h.

I just pushed that fix, all should be ok onow.